app/controllers/settings/imports_controller.rb

```diff
@@ -11,24 +11,27 @@ def show
   end
 
   def create
-    @import = Import.new(import_params)
+    if params[:import].nil?
+      @import = Import.new
+      render :show
+      return
+    end
+
+    @import = Import.new(params[:import].permit(:data, :type))
     @import.account = @account
 
-    if @import.save
-      ImportWorker.perform_async(@import.id)
-      redirect_to settings_import_path, notice: I18n.t('imports.success')
-    else
+    unless @import.save
       render :show
+      return
```

Comment:
Yes, but I think it is the best solution for this case. Do you have any other idea?